draft-thoji-scone-trone-protocol.md

```diff
@@ -310,6 +310,109 @@ This transport parameter is valid for QUIC versions 1 {{QUIC}} and 2
+  Source Connection ID Length (8),
+  Destination Connection ID (0..2040),
+  Destination Connection ID Length (8),
+  Version (32) = 0xTRONE1 or 0xTRONE2,
+  Rate Signal (6) = 0x3F,
+  Reserved (1),
+  Header Form (1) = 1
+}
+~~~
+{: #fig-trone-indication-packet title="TRONE Indication Packet Format"}
+
+A TRONE indication packet is attached directly after a QUIC Initial packet in
+the first UDP datagram. The Rate Signal field MUST be set to 0x3F (63) when
+sent as a TRONE indication packet.
+
+## Sending TRONE Indication
+
+A QUIC endpoint that supports TRONE MAY attach a TRONE indication packet after
+a QUIC Initial packet in the first UDP datagram. This is an opportunistic
+signal that is not guaranteed to be processed by the receiver, as the receiver
+has not yet confirmed its willingness to receive TRONE packets.
+
+When attaching a TRONE indication, the endpoint SHOULD:
+* Place it immediately after the QUIC Initial packet in the datagram
+* Construct the packet with reversed field order as shown in
+  {{fig-trone-indication-packet}}
+* Set the Source Connection ID and Destination Connection ID fields to match
+  those of the QUIC Initial packet
+* Set the Version field to either 0xTRONE1 or 0xTRONE2
+* Set the Header Form bit to 1 (maintaining QUIC invariants)
+* Set the Rate Signal field to 0x3F (63)
+
+QUIC endpoints MUST NOT rely on TRONE indications for correct operation of the
+QUIC protocol or the TRONE protocol. The indication is purely advisory for
+network elements.
+
+## Processing TRONE Indications
+
+Network elements can detect TRONE indication packets using a reverse parsing
+approach, without requiring full parsing of the Initial packet:
+
+1. Identify a UDP datagram containing a QUIC version 1 or 2 packet.
+2. Check if the datagram length exceeds the minimum expected size of a QUIC
+   Initial packet
+3. Examine the last byte of the datagram to see if it has the Header Form bit
+   set to 1
+4. If set, read backward to verify the Version field matches 0xTRONE1 or
+   0xTRONE2
+5. Optionally read the Connection ID fields and match them to the QUIC packet.
+
+The following pseudocode shows how a network element might detect a TRONE
+indication:
+
+~~~ pseudocode
+is_quic = is_quic_datagram(datagram)
+if is_quic and datagram_length > MIN_INITIAL_SIZE:
+  last_byte = datagram[datagram_length - 1]
+
+  if (last_byte & 0x80) == 0x80:
+    version_start = datagram_length - 8
+    version_end = datagram_length - 4
+    potential_version = datagram[version_start:version_end]
+
+    if potential_version == TRONE1_VERSION or potential_version == TRONE2_VERSION:
+      note_potential_trone_support(flow_tuple)
+~~~
+
+This approach allows network elements to detect TRONE support without needing
+to parse the full QUIC Initial packet structure or perform deep packet
+inspection of the TLS handshake.
+
+Network elements that observe TRONE indication packets MAY:
+* Note the UDP 4-tuple for potential future TRONE handling
```

Comment:
It would be helpful for understanding to elaborate more on the implications of these actions? An operator can do whatever it wants, but it is not clear to me what actions we expect them to take in terms of treatment of the flow (besides not trying to extract the SNI)